flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java

```diff
@@ -408,6 +408,79 @@ public static GraphCsvReader fromCsvReader(String edgesPath, ExecutionEnvironmen
 	}
 
 	/**
+	 * Creates a graph from a Adjacency List text file  with Vertex Key values. Edges will be created automatically.
+	 *
+	 * @param filePath a path to an Adjacency List text file with the Vertex data
+	 * @param context  the execution environment.
+	 * @return An instance of {@link org.apache.flink.graph.GraphAdjacencyListReader},
+	 * on which calling methods to specify types of the Vertex ID, Vertex value and Edge value returns a Graph.
+	 */
+	public static GraphAdjacencyListReader fromAdjacencyListFile(String filePath, ExecutionEnvironment context) {
+		return new GraphAdjacencyListReader(filePath, context);
+	}
+
+	/**
+	 * Writes a graph as an Adjacency List formatted text file in a user specified folder.
+	 *
+	 * @param filePath   the path that the Adjacency List formatted text file should be written in
+	 * @param delimiters the delimiters that separate the different value types in the Adjacency List formatted text
+	 *                   file. Delimiters should be provided with the following order:
+	 *                   NEIGHBOR_DELIMITER : separating source from its neighbors
+	 *                   VERTICES_DELIMITER : separating the different neighbors of a source vertex
+	 *                   VERTEX_VALUE_DELIMITER: separating the source vertex-id from the vertex value, as well as the
+	 *                   target vertex-ids from the edge value.
+	 */
+	public void writeAsAdjacencyList(String filePath, String... delimiters) {
+
+		final String NEIGHBOR_DELIMITER = delimiters.length > 0 ? delimiters[0] : "\t";
+
+		final String VERTICES_DELIMITER = delimiters.length > 1 ? delimiters[1] : ",";
+
+		final String VERTEX_VALUE_DELIMITER = delimiters.length > 1 ? delimiters[2] : "-";
```

Comment:
You mean the error in this declaration: 
```java
final String VERTEX_VALUE_DELIMITER = delimiters.length > 1 ? delimiters[2] : "-";
```
and not to check directly for length greater than two, because in that way the user will have to provide all three delimiters or none.